core/orchestrator.go

```diff
@@ -315,16 +316,34 @@ func (n *LivepeerNode) transcodeSeg(config transcodeConfig, seg *stream.HLSSegme
 
 	// Prepare the result object
 	var tr TranscodeResult
+	segHashes := make([][]byte, len(tData))
+	segHashLock := &sync.Mutex{}
+
 	for i, _ := range md.Profiles {
 		if tData[i] == nil {
 			glog.Errorf("Cannot find transcoded segment for %v", seg.SeqNo)
 			continue
 		}
 		tProfileData[md.Profiles[i]] = tData[i]
 		tr.Data = append(tr.Data, tData[i])
+
+		hash := crypto.Keccak256(tData[i])
+		segHashLock.Lock()
+		segHashes[i] = hash
+		segHashLock.Unlock()
 	}
 	os.Remove(fname)
 	tr.OS = config.OS
+
+	if n == nil || n.Eth == nil {
```

Comment:
Could we add unit tests for the following cases?

- Node in off-chain mode should return `tr` with `tr.Sig` empty
- Node in on-chain mode should return `tr` with the correct `tr.Sig`
- Node in on-chain mode that fails to generate a signature on L343 should return `tr` with `tr.Sig = nil` (should be possible to use a mock implementation for the `LivepeerEthClient` interface field on `n`)

[core_test.go](https://github.com/livepeer/go-livepeer/blob/master/core/core_test.go#L85) seems to be the current file for unit tests related to the transcode loop.